site/src/components/mdx/mdx-page-content.tsx

```diff
@@ -146,13 +146,20 @@ export const MdxPageContent: FunctionComponent<MdxPageContentProps> = ({
       ) {
         currentHeading.current = headingAtScrollPosition;
 
-        void router.replace(
-          `${asPath.split("#")[0]}${
-            headingAtScrollPosition.anchor
-              ? `#${headingAtScrollPosition.anchor}`
-              : ""
-          }`,
-        );
+        void router
+          .replace(
+            `${asPath.split("#")[0]}${
+              headingAtScrollPosition.anchor
+                ? `#${headingAtScrollPosition.anchor}`
+                : ""
+            }`,
+            undefined,
+            { shallow: true },
+          )
+          .catch(() => {
+            // Prevents unhandled "Cancel rendering route" error
+            // Happens during fast scrolling between page sections
```

Comment:
Unrelated to the main purpose of the PR, but https://sentry.io/organizations/hashintel/issues/3481546306/?project=6603021&query=is%3Aunresolved (internal link)